srml/staking/src/tests.rs

```diff
@@ -1650,67 +1656,22 @@ fn bond_with_no_staked_value() {
 		assert_eq!(Staking::slot_stake(), 501);
 
 		// no rewards paid to 2 and 4 yet
-		assert_eq!(Balances::free_balance(&2), initial_balance_2);
-		assert_eq!(Balances::free_balance(&4), initial_balance_4);
+		assert_eq!(Balances::free_balance(&2), init_balance_2);
+		assert_eq!(Balances::free_balance(&4), init_balance_4);
 
+		// Compute total payout now for whole duration as other parameter won't change
+		let total_payout_2 = current_total_payout_for_duration(3);
+		assert!(total_payout_2 > 100); // Test is meaningfull if reward something
+		add_reward_points_to_all_elected();
 		start_era(3);
 
-		let reward = Staking::current_session_reward() * 3;
 		// 2 will not get a reward of only 1
 		// 4 will get the rest
-		assert_eq!(Balances::free_balance(&2), initial_balance_2 + 3);
-		assert_eq!(Balances::free_balance(&4), initial_balance_4 + reward - 3);
+		assert_eq!(Balances::free_balance(&2), init_balance_2 + total_payout_2/3/501);
+		assert_eq!(Balances::free_balance(&4), init_balance_4 + 500*total_payout_2/3/501);
 	});
 }
 
-#[test]
-fn bond_with_little_staked_value_bounded_by_slot_stake() {
```

Comment:
this test was relevant when session reward was function of stake but this is no longer the case.

Reply:
Just to give the context: For now, we don't really have any lower limits on how much you can bond. I think bonding with `1 unit` is also possible, which is an extremely small value. while I think there should be a lower bound on this (since bonding is one-way and eats up some storage until.. forever), we don't so I added this test to at least have some understanding of what happens when someone bonds with very little value.

 If the test is checking slot_stake, rewards, those portions should be gone, otherwise, I think the rest of the test _could stay_.

Reply:
brought back